ethcore/src/engines/mod.rs

```diff
@@ -477,4 +470,28 @@ pub mod common {
 		}
 		Ok(())
 	}
+
+	/// Trace rewards on closing block
+	pub fn bestow_block_reward<E: Engine + ?Sized>(block: &mut ExecutedBlock, engine: &E) -> Result<(), Error> {
+		let tracing_enabled = block.tracing_enabled();
+		let fields = block.fields_mut();
+		let mut tracer = ExecutiveTracer::default();
+		// Bestow block reward
+		let reward = engine.params().block_reward;
+		let res = fields.state.add_balance(fields.header.author(), &reward, CleanupMode::NoEmpty)
+			.map_err(::error::Error::from)
+			.and_then(|_| fields.state.commit());
+
+		if tracing_enabled {
```

Comment:
I would inline the variable here `block.tracing_enabled()`
Actually it could even look like this:
```rust
fields.traces.as_mut().map(|mut traces| {
  let mut tracer = ExecutiveTracer::new();
  tracer.trace_reward(...);
  traces.push(tracer.traces())
})
```
and neither the if statement nor the `tracing_enabled` function in block are needed in such case.